cloudinary-core/src/main/java/com/cloudinary/Url.java

```diff
@@ -389,18 +387,13 @@ public String generate(String source) {
 
 
         if (signUrl && (authToken == null || authToken.equals(AuthToken.NULL_AUTH_TOKEN))) {
-            MessageDigest md = null;
-            try {
-                md = MessageDigest.getInstance(longUrlSignature ? "SHA-256" : "SHA-1");
-            } catch (NoSuchAlgorithmException e) {
-                throw new RuntimeException("Unexpected exception", e);
-            }
+            Signer signer = longUrlSignature ? Signer.SHA256 : config.signatureAlgorithm;
```

Comment:
So again, this should be the algorithm enum, not a signer class. And then when it's used in line 396 below, just call the signing method sending it the algorithm.